Source/VaultSharp.Extensions.Configuration/VaultConfigurationSource.cs

```diff
@@ -56,6 +56,6 @@ public VaultConfigurationSource(VaultOptions options, string basePath, string? m
         /// </summary>
         /// <param name="builder">Configuration builder.</param>
         /// <returns>Instance of <see cref="IConfigurationProvider"/>.</returns>
-        public IConfigurationProvider Build(IConfigurationBuilder builder) => new VaultConfigurationProvider(this, this.logger);
+        public virtual IConfigurationProvider Build(IConfigurationBuilder builder) => new VaultConfigurationProvider(this, this.logger);
```

Comment:
Why do you need this method to be virtual? Would you mind sharing an example of code for a child class?

Reply:
Previously i've tried to inherit from the VaultConfigurationProvider class to customize Load method.
To replace the VaultConfigurationProvider with my custom CustomVaultConfigurationProvider i need to inherit from VaultConfigurationSource, but i can't.
so this change is for maintainability - in future i'll be able to customize Load behaviour without PR in VaultSharp.Extensions.Configuration repo

Reply:
I think adding generic AddVaultConfiguration<T>() will solve your problem, because nesting VaultConfigurationSource won't help - it's explicitly used in VaultConfigurationExtensions

Reply:
my customization scenario is 'on the edge of customization' and has internal specifics
so i used 'config.Add<CustomVaultConfigurationSource>()' without VaultConfigurationExtensions class methods.

also, the suggested new api for config.AddVaultConfiguration<TConfigurationSource>() method (with generic Source) is not familiar or commonly used by aspnetcore-devs - feature won't be used in any working production service :) i guess.
so extending the VaultConfigurationExtensions methods for generics - is not popular/usable scenario.